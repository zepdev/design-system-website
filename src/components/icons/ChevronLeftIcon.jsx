import React from 'react'
import PropTypes from 'prop-types'

const ChevronLeftIcon = ({ className, width, height, fill, viewBox, ariaLabel }) => (
  <svg
    width={width}
    height={height}
    xmlns="http://www.w3.org/2000/svg"
    viewBox={viewBox}
    aria-labelledby="title_menu"
    className={className}
  >
    <title id="title_menu">{ariaLabel || 'ChevronLeft'}</title>
    <path fill={fill} d="M20.547 22.107l-6.107-6.107 6.107-6.12-1.88-1.88-8 8 8 8z" />
  </svg>
)

ChevronLeftIcon.propTypes = {
  width: PropTypes.string,
  height: PropTypes.string,
  viewBox: PropTypes.string,
  fill: PropTypes.string,
  ariaLabel: PropTypes.string,
}

ChevronLeftIcon.defaultProps = {
  width: '32',
  height: '32',
  viewBox: '0 0 32 32',
  fill: 'currentColor',
}

export default ChevronLeftIcon
