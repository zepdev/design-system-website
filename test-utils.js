import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from './src/theme'

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
