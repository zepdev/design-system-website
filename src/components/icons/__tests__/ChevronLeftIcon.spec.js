import React from "react"
import { render } from "../../../../test-utils"
import ChevronLeftIcon from "../ChevronLeftIcon"

describe("ChevronLeftIcon", () => {
  it("renders correctly", () => {
    const component = render(<ChevronLeftIcon />)
    expect(component).toMatchSnapshot()
  })
})
