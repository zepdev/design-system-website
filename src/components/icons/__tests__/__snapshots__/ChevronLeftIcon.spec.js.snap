// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChevronLeftIcon renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        aria-labelledby="title_menu"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="title_menu"
        >
          ChevronLeft
        </title>
        <path
          d="M20.547 22.107l-6.107-6.107 6.107-6.12-1.88-1.88-8 8 8 8z"
          fill="currentColor"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      aria-labelledby="title_menu"
      height="32"
      viewBox="0 0 32 32"
      width="32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title
        id="title_menu"
      >
        ChevronLeft
      </title>
      <path
        d="M20.547 22.107l-6.107-6.107 6.107-6.12-1.88-1.88-8 8 8 8z"
        fill="currentColor"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
